backend/src/main/java/com/google/sps/data/MatchPreference.java

```diff
@@ -41,4 +41,17 @@ public static MatchPreference forIntValue(int value) {
     }
     throw new IllegalStateException("Unknown enum value.");
   }
+
+  public static MatchPreference forStringValue(String value) {
+    switch (value) {
+      case "different":
+        return MatchPreference.DIFFERENT;
+      case "any":
+        return MatchPreference.ANY;
+      case "similar":
+        return MatchPreference.SIMILAR;
+      default:
+        return null;
+    }
```

Comment:
There's a good opportunity for a test here. How should this method handle "ANY" or "SiMiLaR"?

Reply:
I added toLowerCase() to value, but also value should always be one of those three strings since that's what the frontend passes, so if it's none of the above, then an error should be thrown

Reply:
Looks good. There's still an opportunity here for a (short) unit test.

Reply:
Added tests